native/acter/src/api/convo.rs

```diff
@@ -313,6 +313,99 @@ impl Convo {
             })
             .await?
     }
+
+    pub async fn msg_draft(&self) -> Result<OptionComposeDraft> {
+        if !self.is_joined() {
+            bail!("Unable to fetch composer draft of a room we are not in");
+        }
+        let room = self.room.clone();
+        RUNTIME
+            .spawn(async move {
+                let draft = room.load_composer_draft().await?;
+
+                Ok(OptionComposeDraft::new(draft.map(|composer_draft| {
+                    let (msg_type, event_id) = match composer_draft.draft_type {
+                        ComposerDraftType::NewMessage => ("new".to_string(), None),
+                        ComposerDraftType::Edit { event_id } => {
+                            ("edit".to_string(), Some(event_id))
+                        }
+                        ComposerDraftType::Reply { event_id } => {
+                            ("reply".to_string(), Some(event_id))
+                        }
+                    };
+                    ComposeDraft::new(
+                        composer_draft.plain_text,
+                        composer_draft.html_text,
+                        msg_type,
+                        event_id,
+                    )
+                })))
+            })
+            .await?
+    }
+
+    pub async fn save_msg_draft(
+        &self,
+        text: String,
+        html: Option<String>,
+        draft_type: String,
+        event_id: Option<String>,
+    ) -> Result<bool> {
+        if !self.is_joined() {
+            bail!("Unable to save composer draft of a room we are not in");
+        }
+        let room = self.room.clone();
+
+        let draft_type = match (draft_type.as_str(), event_id) {
+            ("new", None) => ComposerDraftType::NewMessage,
+            ("edit", id) => {
+                if let Some(id) = id {
+                    ComposerDraftType::Edit {
+                        event_id: OwnedEventId::try_from(id)?,
+                    }
+                } else {
+                    bail!("Invalid event id or not found");
+                }
+            }
+
+            ("reply", id) => {
+                if let Some(id) = id {
+                    ComposerDraftType::Reply {
+                        event_id: OwnedEventId::try_from(id)?,
+                    }
+                } else {
+                    bail!("Invalid event id or not found");
+                }
+            }
```

Comment:
can be simplified in the match...
```suggestion
            ("edit", Some(id)) => ComposerDraftType::Edit {
                        event_id: OwnedEventId::try_from(id)?,
                    },
            ("reply", Some(id)) => ComposerDraftType::Reply {
                        event_id: OwnedEventId::try_from(id)?,
                    },
            ("reply", _)  | ("edit", _) => bail!("Invalid event id");
```